frontend/e2e/$lang+/_public/apply/adult/_adult-flow.spec.ts

```diff
@@ -0,0 +1,247 @@
+import type { Page } from '@playwright/test';
+import { expect, test } from '@playwright/test';
+
+// Reusable funtion to check empty input
+async function hasError(page: Page, error: string) {
+  await expect(page.getByRole('link', { name: error })).toBeVisible();
+}
+
+// Reusable function to fill out date of birth
+async function fillOutDOB(page: Page, year: string) {
+  await page.getByRole('combobox', { name: 'Month' }).selectOption('01');
+  await page.getByRole('textbox', { name: 'Day (DD)' }).fill('1');
+  await page.getByRole('textbox', { name: 'Year (YYYY)' }).fill(year);
+}
+
+// Reusable function to fill out address
+async function fillOutAddress(page: Page, group: string, address: string, unit: string, country: string, province: string, city: string, postalCode: string) {
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'Address', exact: true }).fill(address);
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'Apartment, suite, etc. (optional)', exact: true }).fill(unit);
+  await page.getByRole('group', { name: group }).getByRole('combobox', { name: 'Country', exact: true }).selectOption(country);
+  await page.getByRole('group', { name: group }).getByRole('combobox', { name: 'Province, territory, state, or region', exact: true }).selectOption(province);
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'City or town', exact: true }).fill(city);
+
+  //invalid postal code
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'Postal code or ZIP code', exact: true }).fill('12345678');
+  await page.getByRole('button', { name: 'Continue' }).click();
+  await hasError(page, 'Enter mailing address postal code in the correct format, such as A1A 1A1');
```

Comment:
I don't think this should be here and the `hasError` is specific to `mailing address`.

Reply:
The `submit` action and `hasError` assertion should be done outside of `fillOutAddress`. I think `fillOutAddress` should be refactored like this:

```ts
interface FillOutAddressArgs {
  address: string;
  city: string;
  country: string;
  group: string;
  page: Page;
  postalCode: string;
  province: string;
  unit: string;
}

async function fillOutAddress({ address, city, country, group, page, postalCode, province, unit }: FillOutAddressArgs) {
  const groupLocator = page.getByRole('group', { name: group });
  await groupLocator.isVisible();
  await groupLocator.getByRole('textbox', { name: 'Address', exact: true }).fill(address);
  await groupLocator.getByRole('textbox', { name: 'Apartment, suite, etc. (optional)', exact: true }).fill(unit);
  await groupLocator.getByRole('combobox', { name: 'Country', exact: true }).selectOption(country);
  await groupLocator.getByRole('combobox', { name: 'Province, territory, state, or region', exact: true }).selectOption(province);
  await groupLocator.getByRole('textbox', { name: 'City or town', exact: true }).fill(city);
  await groupLocator.getByRole('textbox', { name: 'Postal code or ZIP code', exact: true }).fill(postalCode);
}
```

Using an object as argument is better in this. It will reduces potential errors and confusion when using it.

Reply:
When called:

```ts
await fillOutAddress({ page, group: 'Mailing address', address: '123 street', unit: '404', country: 'Canada', province: 'Ontario', city: 'Ottawa', postalCode: 'H0H0H0' });
```

Reply:
> The `submit` action and `hasError` assertion should be done outside of `fillOutAddress`. I think `fillOutAddress` should be refactored like this:
> 
> ```ts
> interface FillOutAddressArgs {
>   address: string;
>   city: string;
>   country: string;
>   group: string;
>   page: Page;
>   postalCode: string;
>   province: string;
>   unit: string;
> }
> 
> async function fillOutAddress({ address, city, country, group, page, postalCode, province, unit }: FillOutAddressArgs) {
>   const groupLocator = page.getByRole('group', { name: group });
>   await expect(groupLocator).toBeVisible();
>   await groupLocator.getByRole('textbox', { name: 'Address', exact: true }).fill(address);
>   await groupLocator.getByRole('textbox', { name: 'Apartment, suite, etc. (optional)', exact: true }).fill(unit);
>   await groupLocator.getByRole('combobox', { name: 'Country', exact: true }).selectOption(country);
>   await groupLocator.getByRole('combobox', { name: 'Province, territory, state, or region', exact: true }).selectOption(province);
>   await groupLocator.getByRole('textbox', { name: 'City or town', exact: true }).fill(city);
>   await groupLocator.getByRole('textbox', { name: 'Postal code or ZIP code', exact: true }).fill(postalCode);
> }
> ```
> 
> Using an object as argument is better in this. It will reduces potential errors and confusion when using it.

Also looks cleaner!